backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/VersionInputHandler.java

```diff
@@ -260,63 +261,92 @@ public VersionInputHandler(
       VersioningEntry versioningEntry,
       Map<String, Map.Entry<BlobExpanded, String>> locationBlobWithHashMap,
       String entityId) {
-    return jdbi.useHandle(
-        handle -> {
-          var queryStr =
-              "INSERT INTO versioning_modeldb_entity_mapping "
-                  + " (repository_id, commit, versioning_key, versioning_location, entity_type, versioning_blob_type, blob_hash, "
-                  + entityIdReferenceColumn
-                  + ") "
-                  + " VALUES (:repository_id, :commit, :versioning_key, :versioning_location, :entity_type, :versioning_blob_type, :blob_hash, :entityId)";
-
-          if (versioningEntry.getKeyLocationMapMap().isEmpty()) {
-            Map<String, Object> keysAndParameterMap = new HashMap<>();
-            keysAndParameterMap.put("repository_id", versioningEntry.getRepositoryId());
-            keysAndParameterMap.put("commit", versioningEntry.getCommit());
-            keysAndParameterMap.put("entityId", entityId);
-            keysAndParameterMap.put("entity_type", entity_type);
-            keysAndParameterMap.put("versioning_key", ModelDBConstants.EMPTY_STRING);
-            keysAndParameterMap.put("versioning_location", null);
-            keysAndParameterMap.put("versioning_blob_type", null);
-            keysAndParameterMap.put("blob_hash", null);
-
-            LOGGER.trace("insert experiment run query string: " + queryStr);
-            var query = handle.createUpdate(queryStr);
-
-            // Inserting fields arguments based on the keys and value of map
-            for (Map.Entry<String, Object> objectEntry : keysAndParameterMap.entrySet()) {
-              query.bind(objectEntry.getKey(), objectEntry.getValue());
+    var existingEntityFuture =
+        getVersionedInputs(Collections.singleton(entityId))
+            .thenApply(
+                existingVersioningEntryMap -> existingVersioningEntryMap.get(entityId), executor);
+    return existingEntityFuture.thenCompose(
+        existingVersioningEntry -> {
+          if (existingVersioningEntry != null) {
+            if (existingVersioningEntry.getRepositoryId() != versioningEntry.getRepositoryId()
+                || !existingVersioningEntry.getCommit().equals(versioningEntry.getCommit())) {
+              return InternalFuture.failedStage(
+                  new AlreadyExistsException(
+                      ModelDBConstants.DIFFERENT_REPOSITORY_OR_COMMIT_MESSAGE));
```

Comment:
Do not allowed to update from different repositories and commit (Logic from old code)